src/main/java/edu/cornell/kfs/fp/batch/service/impl/AdvanceDepositServiceImpl.java

```diff
@@ -0,0 +1,1202 @@
+                payerMessage.append("Payer Name was not found for transaction amount $" + getFormattedAmount("##,##,##0.00", transaction.getTransactionAmount()) + " [Date: " + transaction.getEffectiveDate() + "]");
+                payerMessage.append("\n");
+            }
+        }
+
+        return payerMessage.toString();
+    }
+
+    protected boolean validateTimestamp(AchIncomeFile achIncomeFile) {
+        try {
+            getFormattedTimestamp(achIncomeFile, "fileDate/Time");
+        } catch (FormatException e) {
+            LOG.error("Error while validating the ACH Income file: " + e.getMessage());
+            GlobalVariables.getMessageMap().putError(KFSConstants.GLOBAL_ERRORS, KFSKeyConstants.ERROR_CUSTOM, e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    public void sendEmailSummary(List<PhysicalFlatFileInformation> flatFileInformationList) {
+        for (PhysicalFlatFileInformation physicalFlatFileInformation : flatFileInformationList) {
+            List<FlatFileInformation> fileInformations = physicalFlatFileInformation.getFlatFileInfomationList();
+            for (FlatFileInformation fileInformation : fileInformations) {
+                sendEmail(fileInformation);
+            }
+        }
+    }
+
+    private void sendEmail(FlatFileInformation fileInformation) {
+        if (LOG.isDebugEnabled()) {
+            LOG.debug("sendEmail() starting");
+        }
+        MailMessage message = new MailMessage();
+
+        String returnAddress = parameterService.getParameterValueAsString(KFSConstants.CoreModuleNamespaces.FINANCIAL, CuFPConstants.BATCH_DETAIL_TYPE, CuFPConstants.IU_FROM_EMAIL_ADDRESS_PARM_NM);
+        if (StringUtils.isEmpty(returnAddress)) {
+            returnAddress = mailService.getBatchMailingList();
+        }
+        message.setFromAddress(returnAddress);
+        String subject = parameterService.getParameterValueAsString(LoadAchIncomeFileStep.class, CuFPParameterConstants.AchIncome.ACH_INCOME_SUMMARY_EMAIL_SUBJECT);
+
+        message.setSubject(subject);
+        List<String> toAddressList = new ArrayList<>(parameterService.getParameterValuesAsString(LoadAchIncomeFileStep.class, CuFPParameterConstants.AchIncome.ACH_INCOME_SUMMARY_TO_EMAIL_ADDRESSES));
+        message.getToAddresses().addAll(toAddressList);
+        String body = composeAchIncomeSummaryEmailBody(fileInformation);
+        message.setMessage(body);
+
+        try {
+            mailService.sendMessage(message);
+        } catch (InvalidAddressException e) {
+            LOG.error("sendErrorEmail() Invalid email address. Message not sent", e);
+        } catch (MessagingException me) {
+            throw new RuntimeException("Could not send mail", me);
+        }
+
+    }
+
+    private String composeAchIncomeSummaryEmailBody(FlatFileInformation flatFileInformation) {
+        StringBuffer body = new StringBuffer();
```

Comment:
Unless you need to use StringBuffer for thread safety or other reasons, I would suggest using StringBuilder instead for efficiency.